### ISSUE TITLE
Improve how to handle multiple tasks in a dataset

### ISSUE THREAD

**Issue by marcobarilari** [cpp-lln-lab/CPP_SPM#276]
You think / need this code to do **analysis on multiple tasks in the same dataset, ideally this will not delete already processed data (eg task one when running task2) and re-use the same anatomical image**.

Something is missing / unclear about **loop over tasks?**.

You are confused about how to do **keep processed tasks separated in `deivatives/cpp_spm-taskName**.

It is not clear how / why the code does **`bidsCopyRawFolder` deletes previously processed task1 when running task2**.

**Comment by Remi-Gau** [comment]
> Because so far our pipeline only works on a given task, we could make it automatically clear in the output directory name which task it contains

```
opt.derivativesDir = '/Users/barilari/Desktop/data_temp/ses-003_CPP_SPM/derivatives/cpp_spm';
opt.taskName = 'rdk';
opt = setDerivativesDir(opt);
```

would give:

```
derivativesDir: '//Users/barilari/Desktop/data_temp/ses-003_CPP_SPM/derivatives/cpp_spm-task-rdk'
```


> another possibility we can have is that `opt.taskName` can be a cell of strings that lists all the tasks we want to copy and process together.
> could solve some issues:
> - like we don't have to reprocess the anat for each task 
> - and we can make sure that tasks that go together are in the same derivative dataset
> but it can also complicate some of the workflows a bit (we have to loop over tasks)

**Comment by Remi-Gau** [comment]
I think we need to be clear on what we want the pipeline to do:
- we want it to copy and process only the specified task (or taskS) from a raw dataset.
- we want it to not delete stuff that is already in the derivative folder:
  -  so if on day 1 we analyze task A, and on day 2 we analyze task B (and we want those in the same folder), then it should not touch what has already been done.

I think it might be easier / wiser than to make the pipeline too smart (wrt folder naming) and stick close to the BIDS stuff (`derivatives/pipeline-name`) and let the user decide how they want to organize their stuff.

For example, better to have: 
```
task_A/derivatives/cpp-spm/
task_B/derivatives/cpp-spm/
task_C_and_D/derivatives/cpp-spm/
```
where the users those directories in `opt.derivativesDir` for each task or set of tasks rather than
```
derivatives/cpp-spm-task-A/
derivatives/cpp-spm-task-B/
```
where the pipeline creates the folder names itself by trying to append the task name to the folder.

@marcobarilari @CerenB @fedefalag What do you think?

**Comment by Remi-Gau** [comment]
In practice, we need to modify how `bidsCopy` works.

At the moment, it copies each modality folder (`anat`, `dwi`, `func`...) and then it looks in the func folder that was copied for files that do not match the task of interest. 

The way to fix that is to not copy all tasks to start with.

https://github.com/cpp-lln-lab/CPP_SPM/blob/0d9684007d5292c76b45cd892d7ee3a652995e40/src/workflows/bidsCopyRawFolder.m#L123

**Comment by CerenB** [comment]
1. It taskname is not used to create derivatives folder, atm right?
https://github.com/cpp-lln-lab/CPP_SPM/blob/0d9684007d5292c76b45cd892d7ee3a652995e40/src/setDerivativesDir.m

2. with these two options, is it possible with both to have only 1 anat folder and process only once?
3. with both of these options, we do not need to modiy the bidsCopf function, do we? since the pipeline would create different folders for each task's derivatives.

**Comment by Remi-Gau** [comment]
>     1. It `taskname` is not used to create `derivatives` folder, atm right?
>        https://github.com/cpp-lln-lab/CPP_SPM/blob/0d9684007d5292c76b45cd892d7ee3a652995e40/src/setDerivativesDir.m

nope

>     2. with these two options, is it possible with both to have only 1 anat folder and process only once?

that should be easy (if you specify that you want to process several tasks at once) and a bit more complicated if you process task A, and then later task B but want to reuse the same anat (doable but will require a bit more coding).

>     3. with both of these options, we do not need to modify the `bidsCopy` function, do we? since the pipeline would create different folders for each task's derivatives.

well yes we need to modify it because at the moment it can only handle one task and will delete stuff from other tasks.

**Comment by Remi-Gau** [comment]
resolved by #291
